wp-includes/widgets/class-wp-widget-image.php

```diff
@@ -27,65 +27,233 @@ public function __construct() {
+			'style' => 'max-width: 100%; height: auto;',
 		);
-
-		if ( ! $has_caption ) {
+		if ( ! $caption ) {
 			$image_attributes['class'] .= ' align' . $instance['align'];
 		}
+		if ( $instance['alt'] ) {
+			$image_attributes['alt'] = $instance['alt'];
+		}
+
+		$size = $instance['size'];
+		if ( 'custom' === $size || ! has_image_size( $size ) ) {
+			$size = array( $instance['width'], $instance['height'] );
+		}
+
+		$image = wp_get_attachment_image( $attachment->ID, $size, false, $image_attributes );
 
-		$image = wp_get_attachment_image( $attachment->ID, $instance['size'], false, $image_attributes );
 		$url = '';
-		if ( 'file' === $instance['link'] ) {
+		if ( 'file' === $instance['link_type'] ) {
 			$url = wp_get_attachment_url( $attachment->ID );
-		} elseif ( 'post' === $instance['link'] ) {
+		} elseif ( 'post' === $instance['link_type'] ) {
 			$url = get_attachment_link( $attachment->ID );
-		} elseif ( 'custom' === $instance['link'] && ! empty( $instance['link_url'] ) ) {
+		} elseif ( 'custom' === $instance['link_type'] && ! empty( $instance['link_url'] ) ) {
 			$url = $instance['link_url'];
 		}
 
 		if ( $url ) {
-			$image = sprintf( '<a href="%s">%s</a>', esc_url( $url ), $image );
+			$image = sprintf(
+				'<a href="%1$s" class="%2$s" rel="%3$s" target="%4$s">%5$s</a>',
+				esc_url( $url ),
+				esc_attr( $instance['link_classes'] ),
+				esc_attr( $instance['link_rel'] ),
+				! empty( $instance['link_target_blank'] ) ? '_blank' : '',
+				$image
+			);
 		}
 
-		if ( $has_caption ) {
+		if ( $caption ) {
 			$width = 0;
 			$size = _wp_get_image_size_from_meta( $instance['size'], wp_get_attachment_metadata( $attachment->ID ) );
 			if ( false !== $size ) {
 				$width = $size[0];
 			}
 			$image = img_caption_shortcode( array(
-				'id'      => $widget_id . '-caption',
-				'width'   => $width,
-				'align'   => 'align' . $instance['align'],
-				'caption' => $attachment->post_excerpt,
+				'width' => $width,
+				'align' => 'align' . $instance['align'],
+				'caption' => $caption,
 			), $image );
 		}
 
 		echo $image;
 	}
+
+	/**
+	 * Loads the required media files for the media manager and scripts for .
+	 *
+	 * @since 4.8.0
+	 * @access public
+	 */
+	public function enqueue_admin_scripts() {
+		parent::enqueue_admin_scripts();
+
+		$handle = 'media-image-widget';
+		wp_enqueue_script( $handle );
+
+		wp_add_inline_script(
+			$handle,
+			sprintf(
+				'wp.mediaWidgets.modelConstructors[ %s ].prototype.defaults = %s;',
+				wp_json_encode( $this->id_base ),
+				wp_json_encode( $this->default_instance )
+			)
+		);
+
+		wp_add_inline_script(
+			$handle,
+			sprintf(
+				'
+					wp.mediaWidgets.controlConstructors[ %1$s ].prototype.mime_type = %2$s;
+					_.extend( wp.mediaWidgets.controlConstructors[ %1$s ].prototype.l10n, %3$s );
+				',
+				wp_json_encode( $this->id_base ),
+				wp_json_encode( $this->widget_options['mime_type'] ),
+				wp_json_encode( $this->l10n )
+			)
+		);
```

Comment:
@obenland yes, by using `wp_add_inline_script()` it means we can avoid having to create a new global variable and can instead amend an existing object directly. Also, `wp_localize_script()` will cast scalar array values to strings which may not always be desired.